tests/apollo/test_skvbc_state_transfer.py

```diff
@@ -113,56 +114,54 @@ async def test_state_transfer_with_multiple_clients(self, bft_network,exchange_k
         await skvbc.assert_successful_put_get()
         await bft_network.force_quorum_including_replica(stale_node)
         await skvbc.assert_successful_put_get()
+    
 
-    # This test should never run with TLS/TCP, only UDP
     @skip_for_tls
     @with_trio
     @with_bft_network(start_replica_cmd,
                     selected_configs=lambda n, f, c: f >= 2,
                     rotate_keys=True)
     async def test_state_transfer_with_internal_cycle(self, bft_network, exchange_keys=True):
         """
-        state transfer starts and destination replica(fetcher) is
+        State transfer starts and destination replica (fetcher) is
         blocked during getMissingBlocks state, a bunch of data is added
-        to the rest of the cluster  using multiple clients. Then destination
-        replica is unblocked and state transfer resume . After one cycle
-        of state transfer is done replica goes into another internal(
-        from getReserved page state to getCheckpointSummary State)
-        cycle to fetch remaining blocks which were added during first cycle.
+        to the rest of the cluster using multiple clients. Then destination
+        replica is unblocked and state transfer resumes. After one cycle
+        of state transfer is done replica goes into another internal (
+        from getReserved page state to getCheckpointSummary State) cycle
+        to fetch remaining blocks which were added during first cycle.
         """
 
         skvbc = kvbc.SimpleKVBCProtocol(bft_network)
         stale_node = random.choice(
             bft_network.all_replicas(without={0}))
+        stale_nodes={stale_node}
 
-        await skvbc.start_replicas_and_write_with_multiple_clients(
+        await skvbc.prime_for_state_transfer(
             stale_nodes={stale_node},
-            write_run_duration=30,
+            checkpoints_num=4,
             persistency_enabled=False
         )
-
-        with ntc.NetworkTrafficControl() as tc:
-            #delay added to loopback interface to capture the stale replica in GettingMissingBlocks state
-            tc.put_loop_back_interface_delay(200)
+        
+        with net.ReplicaSubsetTwoWayIsolatingAdversary(bft_network, stale_nodes) as adversary:
             bft_network.start_replica(stale_node)
-            with trio.fail_after(30):
-                while(True):
-                    fetchingState = await bft_network.get_metric(stale_node, bft_network,
-                                                             "Statuses", "fetching_state", "bc_state_transfer")
-                    if(fetchingState == "GettingMissingBlocks"):
-                        break
-                    await trio.sleep(0.1)
-        await skvbc.send_concurrent_ops_with_isolated_replica(isolated_replica={stale_node}, run_duration=30)
+            adversary.interfere(10)
```

Comment:
1)
Can you please elaborate? I think you should consider adding a comment to explain what you are doing here in this whole scope?
It is not clear when one try to read fast, spending minimal time and avoid digging into the code.
Will you be able to understand in 5 seconds what is done here 6 months from now?

2)
adversary.interfere(10) means? I can't see a single interfere() call in all tests with a number. Why this can even run?

3)
In addition, How can we be sure that replica reaches getting missing blocks when interfere is called?
I understanf that you ran it 50 times, but i'm not sure if it was ran 50 times on remote. In addition, It might fail once in 150 times (for example) in avarage , as the methodology looks opportunistic.